### DISCUSSION TITLE
vCPU hours for Amazon RDS Proxy

### DISCUSSION THREAD

**Discussion by nshelia** [cloud-carbon-footprint/cloud-carbon-footprint#1129] [Q&A] [answered]
Hey, 

I have a script which takes a billing report ( Amazon RDS ) and calculates emissions based on Amazon RDS service and usage type which is **Proxy-ASv2-Usage** in my case. 

For some reason I am getting NaN from [method defined here](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/trunk/packages/aws/src/lib/CostAndUsageReportsRow.ts#L110). Presumably, there is no direct mapping to vCPU hours like there is for [EC2 instance types](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/trunk/packages/aws/src/lib/CostAndUsageReportsRow.ts#L116). 

Question: can we somehow approximate vCPU hours for Amazon RDS Proxy ?

**Comment by 4upz** [comment, marked as answer]
Hi @nshelia,

Thanks for your question! It seems that Amazon RDS Proxy is a managed service. Therefore, based on the usage type, it would be classified and calculated according to our [unknown methodology](https://www.cloudcarbonfootprint.org/docs/classifying-usage-types/#handling-unknown-usage-types).

We can add support for this, however, we would need additional information such as how this service would appear as a line item on a billing report. If you're willing to provide an example of this data, feel free to do so (with any sensitive info masked/changed).

Alternatively, you can refer to the [Pricing Page](https://aws.amazon.com/rds/proxy/pricing/?nc=sn&loc=3) and at least use the minimum provisioned vCPU value of 2 as a place to start. Just keep in mind that since this is a managed service, this could result in actual emissions being underestimated due to the potential of it scaling up its provisioned instances to match your usage.
